contracts/bonding/BondingManager.sol

```diff
@@ -248,44 +224,46 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
             del.startRound = currentRound.add(1);
             // Unbonded state = no existing delegate and no bonded stake
             // Thus, delegation amount = provided amount
-        } else if (del.delegateAddress != address(0) && _to != del.delegateAddress) {
+        } else if (currentDelegate != address(0) && currentDelegate != _to) {
             // A registered transcoder cannot delegate its bonded stake toward another address
             // because it can only be delegated toward itself
             // In the future, if delegation towards another registered transcoder as an already
             // registered transcoder becomes useful (i.e. for transitive delegation), this restriction
             // could be removed
-            require(transcoderStatus(msg.sender) == TranscoderStatus.NotRegistered);
+            require(!isRegisteredTranscoder(msg.sender), "registered transcoders can't delegate towards other addresses");
             // Changing delegate
             // Set start round
             del.startRound = currentRound.add(1);
             // Update amount to delegate with previous delegation amount
             delegationAmount = delegationAmount.add(del.bondedAmount);
-            // Decrease old delegate's delegated amount
-            delegators[currentDelegate].delegatedAmount = delegators[currentDelegate].delegatedAmount.sub(del.bondedAmount);
 
-            if (transcoderStatus(currentDelegate) == TranscoderStatus.Registered) {
+            if (isRegisteredTranscoder(currentDelegate)) {
                 // Previously delegated to a transcoder
                 // Decrease old transcoder's total stake
-                transcoderPool.updateKey(currentDelegate, transcoderTotalStake(currentDelegate).sub(del.bondedAmount), address(0), address(0));
+                decreaseTranscoderTotalStake(currentDelegate, del.bondedAmount);
             }
+
+            // Decrease old delegate's delegated amount
+            delegators[currentDelegate].delegatedAmount = delegators[currentDelegate].delegatedAmount.sub(del.bondedAmount);
         }
 
         // Delegation amount must be > 0 - cannot delegate to someone without having bonded stake
-        require(delegationAmount > 0);
+        require(delegationAmount > 0, "delegation amount must be greater than 0");
         // Update delegate
         del.delegateAddress = _to;
-        // Update current delegate's delegated amount with delegation amount
-        delegators[_to].delegatedAmount = delegators[_to].delegatedAmount.add(delegationAmount);
-
-        if (transcoderStatus(_to) == TranscoderStatus.Registered) {
+        // Update bonded amount
+        del.bondedAmount = del.bondedAmount.add(_amount);
```

Comment:
Otherwise if we would bond from being unbonded we would always count as inactive and never try to join the active set because we only update the bonded amount after calling that statement.

Reply:
Otherwise if we would bond from being unbonded we would always count as inactive and never try to join the active set because we only update the bonded amount after calling that statement.